internals/daemon/daemon.go

```diff
@@ -58,6 +58,66 @@ var (
 	systemdSdNotify = systemd.SdNotify
 )
 
+// TransportType defines the possible API transport types we support. The
+// type can be extracted from the http.Request using RequestTransportType.
+type TransportType int
+
+// Used with context.WithValue as the key.
+type TransportTypeKey struct{}
+
+const (
+	TransportTypeUnknown TransportType = iota // Must be zero value => 0
+	TransportTypeUnixSocket
+	TransportTypeHTTP
+	TransportTypeHTTPS
+)
+
+// RequestTransportType extracts the transport type of the HTTP request. If
+// the transport cannot be found in the context, it returns the zero value
+// TransportTypeUnknown.
+func RequestTransportType(r *http.Request) TransportType {
+	if r == nil {
+		return TransportTypeUnknown
+	}
+	// If the assertion fails here, transport will always be the zero
+	// value of TransportType (TransportTypeUnknown).
+	transport, _ := r.Context().Value(TransportTypeKey{}).(TransportType)
+	return transport
+}
+
+// String returns a string representation of the transport type.
+func (t TransportType) String() string {
+	switch t {
+	case TransportTypeUnixSocket:
+		return "local"
+	case TransportTypeHTTP:
+		return "HTTP"
+	case TransportTypeHTTPS:
+		return "HTTPS"
+	default:
+		return "unknown"
+	}
+}
+
+// IsSupported returns true of the transport type is not unknown.
+func (t TransportType) IsSupported() bool {
+	switch t {
+	case TransportTypeUnixSocket, TransportTypeHTTP, TransportTypeHTTPS:
+		return true
+	}
+	return false
+}
+
+// IsSafe returns true if the transport type is either encrypted (HTTPS) or
+// if its local to the device (Unix Domain Socket).
+func (t TransportType) IsSafe() bool {
```

Comment:
nit: maybe `IsSecure` or `IsTrusted`?

Reply:
Maybe there is a better word, but I would argue it is not secure on its own, not without trust of the thing you are talking to, and equally it does not imply trust. Let me think a bit more ...

Reply:
I'd just go with `IsSafe` then!

Reply:
Yeah, IsSafe is not terrible. `IsLocalOrEncrypted` is accurate but rather a mouthful.

Reply:
Changed to ```IsConcealed```. Resolving.